File: 0049-group-anagrams/0049-group-anagrams.py
from collections import defaultdict
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        my_dict=defaultdict(list)
        for s in strs:
            sorted_s=tuple(sorted(s))
            my_dict[sorted_s].append(s)
        result=[]
        for value in my_dict.values():
            result.append(value)
        return result

File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        map<string,vector<string>>hash;
        for(int i=0;i<strs.size();i++){
            string s=strs[i];
            sort(s.begin(),s.end());
            if(hash.find(s)!=hash.end()) hash[s].push_back(strs[i]);
            else hash[s]={strs[i]};
        }
        vector<vector<string>>res;
        for(auto i:hash) res.push_back(i.second);
        return res;
    }
};